docs/configuration.md

```diff
@@ -629,3 +629,104 @@ window.$docsify = {
   topMargin: 90, // default: 0
 };
 ```
+
+## vueComponents
+
+- type: `Object`
+
+Registers Vue components using the component name as the key with an object containing Vue options as the value.
+
+```js
+window.$docsify = {
+  vueComponents: {
+    'button-counter': {
+      template: `
+        <button @click="count += 1">
+          You clicked me {{ count }} times
+        </button>
+      `,
+      data() {
+        return {
+          count: 0,
+        };
+      },
+    },
+  },
+};
+```
+
+```markdown
+<button-counter></button-counter>
+```
+
+<output data-lang="output">
+  <button-counter></button-counter>
+</output>
+
+## vueGlobalOptions
+
+- type: `Object`
+
+Specifies Vue options to be shared throughout your site. These options will be used when Docsify detects Vue content in the main content area that has not been previously mounted via [vueMountOptions](#vuemountoptions), [vueComponents](#vuecomponents), or a markdown `<script>`.
+
+```js
+window.$docsify = {
+  vueGlobalOptions: {
+    data() {
+      return {
+        count: 0,
+      };
+    },
+  },
+};
+```
+
+```markdown
+<p>
+  <button v-on:click="count -= 1">-</button>
+  {{ count }}
+  <button v-on:click="count += 1">+</button>
+</p>
+```
+
+<output data-lang="output">
+  <p>
+    <button v-on:click="count -= 1">-</button>
+    {{ count }}
+    <button v-on:click="count += 1">+</button>
+  </p>
+</output>
+
+## vueMountOptions
+
+- type: `Object`
+
+Specifies Vue mount elements and their associated options. Mount elements are specified using a [CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) as the key with an object containing Vue options as their value.
+
+```js
+window.$docsify = {
+  vueMountOptions: {
+    '#counter': {
+      data() {
+        return {
+          count: 0,
+        };
+      },
+    },
+  },
+};
+```
```

Comment:
I feel that there's explanation missing here as to what exactly this does. 

If the vueMountOptions contains 5 selectors, does that mean Vue instantiates 5 components and mounts to those five areas on the page?

Reply:
@trusktr Agreed. Good catch. This question is answered on the `vue.md` page, but I neglected to add it to the description here as well.

> Docsify will mount the first matching element in the main content area (`#main, .markdown-section`) each time a new page is loaded.

For example:

```js
window.$docsify = {
  vueMountOptions: {
    // The CSS selector "p" could match multiple elements.
    // However, only the first <p> element will be mounted.
    'p': {  
      data() {
        return {
          count: 0,
        };
      },
    },
  },
};
```

Reply:
Why not have it apply to all matches? That would be convenient I imagine (one definition, but multiple instances of it across the page). Thinking `querySelectorAll` instead of `querySelector`.

Also people may wonder if the following examples will work:

**Exampe 1**

Does it render both?

```html
<p class="one">{count}</p>
<p class="two">{count}</p>
```

```js
window.$docsify = {
  vueMounts: {
    'p.one': {  
      data: () => ({ count: 1 },
    },
    'p.two': {  
      data: () => ({ count: 2 },
    },
  },
};
```

**Example 2**

What happens?

```html
<p class="one">
  {count}
  <p class="two">
    {count}
  </p>
</p>
```

```js
window.$docsify = {
  vueMounts: {
    'p.one': {  
      data: () => ({ count: 1 },
    },
    'p.two': {  
      data: () => ({ count: 2 },
    },
  },
};
```

**Example 3**

What happens?

```html
<component-one inline-template>
  {count}
  <component-two inline-template>
    {count}
  </component-two>
</component-one>
```

```js
window.$docsify = {
  vueComponents: {
    'component-one': {
      data: () => ({ count: 1 },
    },
    'component-two': {  
      data: () => ({ count: 2 },
    },
  },
};
```

**Example 4**

User opens feature request saying "can we render all of them?"

```html
<p>{count}</p>
<p>{count}</p>
<p>{count}</p>
<p>{count}</p>
<p>{count}</p>
```

```js
let pCount = 0

window.$docsify = {
  vueMounts: {
    'p': {  
      data: () => ({ count: ++pCount  },
    }
  },
};
```

**Exampe 5**

How do I make this work?

```html
<p class="one">{count}{foo}</p>
<p class="two">{count}{foo}</p>
```

```js
window.$docsify = {
  vueMounts: {
    'p.one': {
      data: () => ({ count: 1 },
    },
    'p.two': {
      data: () => ({ count: 2 },
    },
    'p': {
      data: () => ({ foo: 'bar' },
    },
  },
};
```